bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SortedLedgerStorage.java

```diff
@@ -41,16 +42,23 @@
  * entries will be first added into a {@code MemTable}, and then be flushed back to the
  * {@code InterleavedLedgerStorage} when the {@code MemTable} becomes full.
  */
-public class SortedLedgerStorage extends InterleavedLedgerStorage
-        implements LedgerStorage, CacheCallback, SkipListFlusher {
+public class SortedLedgerStorage
```

Comment:
this is a important change, SortedLedgerStorage will not be a InterleavedLedgerStorage anymore

cc @sijie @ivankelly  @merlimat

Reply:
What are the implications of this change? @eolivelli ?

Reply:
@dlg99 can you please describe the motto of this change?

Reply:
This simplifies testability i.e. SlowSortedLedgerStorage implementation. Reduces code duplication etc. I tried dealing with this without this change first and turn up ugly.

I am not worried about this as there is no reason to code directly against InterleavedLedgerStorage instead of CompactableLedgerStorage interface which SortedLedgerStorage still implements. Tests still pass though I had to change couple of tests that casted SortedLedgerStorage to InterleavedLedgerStorage. 
If there is some specific reason that I have missed that explains why we should not change that or a reason why we should ignore interface and code against concrete implementation please let me know.

Reply:
@dlg99 thank you for trying.
Go head with your change.
We already told at community meeting that this is not a problem